mycroft/stt/deepspeech_stt.py

```diff
@@ -0,0 +1,131 @@
+        except ImportError:
+            LOG.error("could not import deep speech, \n run pip install "
+                      "deepspeech")
+            raise
+
+        self.BEAM_WIDTH = self.config.get("beam_width", 500)
+        self.LM_WEIGHT = self.config.get("lm_weight", 1.75)
+        self.WORD_COUNT_WEIGHT = self.config.get("word_count_weight", 1.00)
+        self.VALID_WORD_COUNT_WEIGHT = self.config.get(
+            "valid_word_count_weight", 1.00)
+
+        # These constants are tied to the shape of the graph used
+        # (changing them changes the geometry of the first layer), so make
+        # sure you use the same constants that were used during training
+        self.N_FEATURES = self.config.get("n_features", 26)
+        self.N_CONTEXT = self.config.get("n_context", 9)
+
+        self.model = self.config.get("model", join(dirname(__file__),
+                                                   "deepspeech",
+                                                   "output_graph.pb"))
+        self.lm = self.config.get("lm", join(dirname(__file__),
+                                             "deepspeech",
+                                             "lm.binary"))
+        self.alphabet = self.config.get("alphabet", join(dirname(__file__),
+                                                         "deepspeech",
+                                                         "alphabet.txt"))
+        self.trie = self.config.get("trie", join(dirname(__file__),
+                                                 "deepspeech",
+                                                 "trie"))
+        self.downloaded = False
+        self.dl = None
+        auto_dl = self.config.get("auto_download", False)
+        if self.is_ready(not auto_dl):
+            self.load_model()
+        elif auto_dl:
+            LOG.info("Downloading model")
+            self.download()
+            while not self.downloaded:
+                sleep(1)
+
+            if self.is_ready(True):
+                self.load_model()
+
+    def load_model(self):
+        LOG.info('Loading model from file %s' % (self.model))
+        model_load_start = timer()
+        self.ds = self.Model(self.model, self.N_FEATURES, self.N_CONTEXT,
+                             self.alphabet, self.BEAM_WIDTH)
+
+        model_load_end = timer() - model_load_start
+        LOG.debug('Loaded model in %0.3fs.' % (model_load_end))
+
+        LOG.info('Loading language model from files %s %s' % (
+            self.lm, self.trie))
+        lm_load_start = timer()
+        self.ds.enableDecoderWithLM(self.alphabet, self.lm, self.trie,
+                                    self.LM_WEIGHT,
+                                    self.WORD_COUNT_WEIGHT,
+                                    self.VALID_WORD_COUNT_WEIGHT)
+        lm_load_end = timer() - lm_load_start
+        LOG.debug('Loaded language model in %0.3fs.' % (lm_load_end))
+
+    def is_ready(self, is_critical=False):
+        try:
```

Comment:
I'm not a fan of this pattern of throwing, catching, and conditionally rethrowing. Why don't we just return success or failure and then throw at the caller if necessary?

Reply:
this way we know what failed in more detail, i will throw a generic model not found exception instead